core/network/network.go

```diff
@@ -61,10 +61,13 @@ const (
 	// CannotConnect means recently attempted connecting but failed to connect.
 	// (should signal "made effort, failed")
 	CannotConnect
+
+	// Transient means we have a transient connection to the peer, but aren't fully connected.
+	Transient
```

Comment:
Call this something else?

Reply:
I prefer `limited`. 
This is fine too. At least it's consistent with the network.UseTransient.

Reply:
relayed?

Reply:
Limited/Temporary make the most sense to me. It may be relayed, it may not be.

Reply:
I prefer `Limited` because the opposite `Unlimited` also makes sense. I don't think `Temporary` / `Permanent` conveys that correctly. Because whether a connection is transient or not depends on how much data we can transfer over it. Relayed connections with no data limit count as Nontransient.

Reply:
I would prefer we rename this while we are here. `Limited` sounds good to me.

After a couple of months of not working on go-libp2p, it took me a while to remember what a "transient" connection really way. Transient implies that the connection may disappear soon. Which may not be the case

Reply:
Limited sounds like the way to go then.

Reply:
Renamed to `Limited`